### ISSUE TITLE
Question/bug : Zipkin visualize by client send/client receive and missing server events

### ISSUE THREAD

**Issue by tdanylchuk** [openzipkin/zipkin#2145]
Not sure it's a bug, but a question for sure.
I have multiple services env and want to trace request between them.
The data flow is following:
1. `zuul-server` calls `camel-producer` via rest
2. `camel-producer` sends event to some queue (I use kafka in my case)
3. `camel-consumer` receives event from queue and calls `feign-service` via rest

I simulate lag, by not starting consumer, but starting it only after message was sent, so expect to see this lag on zipkin, but instead of it I got:
![image](https://user-images.githubusercontent.com/3741358/43002220-ad942fce-8c30-11e8-84d6-67b67e4d321a.png)
The problem here is `camel-consumer` section, it renders by client events from `camel-producer` and skips server events from `camel-consumer`. Events from `camel-consumer` section:
![image](https://user-images.githubusercontent.com/3741358/43002301-ed6a25cc-8c30-11e8-9d23-77d9e58f5d92.png)

Here is JSONs which were sent during the invocation:
```
[
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "2b99f30e57a93451",
    "id": "b79cd0676def9844",
    "kind": "CLIENT",
    "name": "camel::direct://newcamel",
    "timestamp": 1532083449680000,
    "duration": 179000,
    "localEndpoint": {
      "serviceName": "camel-producer",
      "ipv4": "11.11.11.111",
      "port": 8882
    },
    "tags": {
      "camel_span": "true"
    }
  }
]
```
```
[
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "6a3fa724666d301d",
    "id": "2b99f30e57a93451",
    "kind": "CLIENT",
    "name": "http:/camel-producer/newcamel",
    "timestamp": 1532083449475000,
    "duration": 3875000,
    "localEndpoint": {
      "serviceName": "zuul-server",
      "ipv4": "11.111.111.111",
      "port": 8881
    },
    "tags": {
      "http.method": "GET",
      "http.path": "/newCamel",
      "http.status_code": "200",
      "http.url": "/newCamel",
      "lc": "zuul"
    }
  },
  {
    "traceId": "6a3fa724666d301d",
    "id": "6a3fa724666d301d",
    "kind": "SERVER",
    "name": "http:/camel-producer/newcamel",
    "timestamp": 1532083449473000,
    "duration": 3880456,
    "localEndpoint": {
      "serviceName": "zuul-server",
      "ipv4": "111.111.111.111",
      "port": 8881
    },
    "tags": {
      "http.host": "localhost",
      "http.method": "GET",
      "http.path": "/camel-producer/newCamel",
      "http.status_code": "200",
      "http.url": "http://localhost:8881/camel-producer/newCamel"
    }
  }
]
```
```
[
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "6a3fa724666d301d",
    "id": "2b99f30e57a93451",
    "kind": "SERVER",
    "name": "http:/camel-producer/newcamel",
    "timestamp": 1532083449620000,
    "duration": 3728071,
    "localEndpoint": {
      "serviceName": "camel-producer",
      "ipv4": "11.11.111.111",
      "port": 8882
    },
    "tags": {
      "mvc.controller.class": "ExampleClientApplication",
      "mvc.controller.method": "newCamel"
    },
    "shared": true
  }
]
```
```
[
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "b79cd0676def9844",
    "id": "ffdba7a51db98e75",
    "kind": "CLIENT",
    "name": "http:/getconfiginfo",
    "timestamp": 1532083480051000,
    "duration": 164000,
    "localEndpoint": {
      "serviceName": "camel-consumer",
      "ipv4": "10.63.22.158",
      "port": 8080
    },
    "tags": {
      "http.method": "GET",
      "http.path": "/getConfigInfo"
    }
  },
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "2b99f30e57a93451",
    "id": "b79cd0676def9844",
    "kind": "SERVER",
    "name": "camel::direct://newcamel",
    "timestamp": 1532083479566000,
    "duration": 674000,
    "localEndpoint": {
      "serviceName": "camel-consumer",
      "ipv4": "111.11.11.11",
      "port": 8080
    },
    "tags": {
      "camel_span": "true"
    }
  }
]
```
```
[
  {
    "traceId": "6a3fa724666d301d",
    "parentId": "b79cd0676def9844",
    "id": "ffdba7a51db98e75",
    "kind": "SERVER",
    "name": "http:/getconfiginfo",
    "timestamp": 1532083480179000,
    "duration": 39506,
    "localEndpoint": {
      "serviceName": "feign-service",
      "ipv4": "11.111.11.111",
      "port": 9093
    },
    "tags": {
      "mvc.controller.class": "ConfigClientController",
      "mvc.controller.method": "getConfigInfo",
    },
    "shared": true
  }
]
```

**Comment by codefromthecrypt** [comment]
in this case the camel instrumentation needs to be updated. The problem is that it is treating everything like a client-server request, but also I'm not sure the lags agree with what you expect.

The same thing that red hat folks have done to update the opentracing tracer could be done to the zipkin one, and likely things would be fine. opentracing api is very similar to brave 4.

https://github.com/apache/camel/blob/master/components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/AbstractMessagingSpanDecorator.java

I've helped with renovating as red hat folks for whatever reason don't lately. It would be nice if someone could take a stab at porting what's been done in camel-opentracing to camel-zipkin with brave 4. The current code in camel-zipkin is basically dead end and if anyone (regardless of employer) were to take the basics of what red hat did in camel-opentracing back to the original tracer, I suspect a lot of folks would be better off for it. Could you be that person?

**Comment by codefromthecrypt** [comment]
I opened https://issues.apache.org/jira/browse/CAMEL-12672 to solve the root problem, which is that the instrumentation approach is dead-end

**Comment by tdanylchuk** [comment]
Thank you so much for your investigation and time you've spent, but unfortunately I messed everything up with service naming and led you to wrong assumptions. 
`camel-producer` and `camel-consumer` are not using `camel-zipkin` lib, that lib is not covering my needs, so I decided to make lib myself (https://github.com/Playtika/sleuth-camel) hopeful after I get proper result we will publish it to maven central. Next step - to contribute to spring-cloud-sleuth project for 2.0 boot and brave support, my lib is targeting 1.5 boot which is my preference so far.

Anyway issue I posted is related to JSON's which I attached below, they seems to be correct and I see that Zipkin got them and parse them in proper way, but unfortunately not showing the desired result of lag, which I trying yo achieve. I'm not pointing that this is a bug, maybe it's by design, just wanted to have confirmation of that.
